provisioning/device/src/ProvisioningTransportRegisterRequest.cs

```diff
@@ -56,6 +61,26 @@ public string ProductInfo
             GlobalDeviceEndpoint = globalDeviceEndpoint;
             IdScope = idScope;
             Security = security;
+            Data = null;
+        }
+
+        /// <summary>
+        /// Creates a new instance of the ProvisioningTransportRegisterMessage class.
+        /// </summary>
+        /// <param name="globalDeviceEndpoint">The Global Device Endpoint for this message.</param>
+        /// <param name="idScope">The IDScope for this message.</param>
+        /// <param name="security">The SecurityProvider used to authenticate the client.</param>
+        /// <param name="data">The custom content.</param>
+        public ProvisioningTransportRegisterMessage(
+            string globalDeviceEndpoint,
+            string idScope,
+            SecurityProvider security,
+            TwinCollection data)
+        {
```

Comment:
Why is twin collection input parameter to user API (as it is public)?

Reply:
This was relevant to the Register input changes, but e do not need Register input here or anywhere. I'll remove it